cli/2024-04-azure-workload-identity.md

```diff
@@ -0,0 +1,233 @@
+
+1. **Managed Identities** are a special type of service principal that eliminate the need for users to manage credentials by providing an identity for the Azure resource in Azure AD and using it to obtain Azure Active Directory (Azure AD) tokens. These can be used only with the scope of Azure ecosystem. There are two types of managed identities:
+    * **System assigned managed identities**: This managed identity is tied to the lifecycle of the Azure resource and are used to access a particular resource.
+    * **User assigned managed identities**: This managed identity is created as a separate resource or entity in Azure and are assigned to multiple Azure resources.
+
+1. **Workload identities** use the same service principal or managed identity to trust tokens from an external identity provider (IdP). Once that trust relationship is created, your external software workload exchanges trusted tokens from the external IdP for access tokens from the Microsoft Identity platform. Your software workload uses that access token to access the protected Azure resources to which the workload has been granted access. This eliminates the burden of managing and rotating the secrets.
+
+1. **Azure AD Workload Identity mutating admission webhook (AzWI mutating admission webhook)** is a pod on the Kubernetes cluster that intercepts the pod creation request and injects the signed service account token into the pod. This signed service account token is used by the Azure provider to authenticate with Azure resources.
+
+
+## Objectives
+
+> **Issue Reference: https://github.com/radius-project/radius/issues/7308**
+
+### Goals
+
+* Radius users can configure Azure provider to use Workload Identity for authentication.
+* Workload Identity can be configured via interactive experience.
+* Workload Identity can be configured manually.
+* Radius users can deploy and manage Azure resources without using service principal credentials.
+
+### Non-goals
+
+* Azure Managed Identity support
+* AWS Workload Identity (IRSA) support
+
+### User scenarios
+
+#### User story 1: As an infrastructure operator, I can configure the Radius Azure Provider to use Workload Identity for authentication via interactive experience.
+
+1. Install the `rad` CLI.
+
+1. run `rad init --full` to begin the interactive experience
+    ```
+    rad init --full
+    ```
+
+1. Enter the environment name, namespace and select "yes" to add a cloud provider and select "Azure" to add an Azure cloud provider.
+    ```
+    Select your cloud provider:
+    > 1. Azure
+      2. AWS
+      3. [back]
+    ```
+
+1. Enter or use the Azure subscription and resource group name. Select  "Workload Identity" for the Azure cloud provider.
+    ```
+    Select the identity for the Azure cloud provider 
+    1. Service principal                               
+    > 2. Workload Identity    
```

Comment:
If Azure is removing support for secrets for internal apps, how are we going to validate service principals still work for our users?